### ISSUE TITLE
SDL responses with GENERIC_ERROR instead of UNSUPPORTED_RESOURCE

### ISSUE THREAD

**Issue by KhrystynaDubovyk** [smartdevicelink/sdl_core#1006]
## Occurrence
Always.

## Preconditions
SDL is started.

## Steps to reproduce
Send any single UI-related RPC , UI interface is not supported by the system (script.lua from attachment could be used to reproduce)
SDL receives UI.IsReady (available=false) from HMI

## Actual result
SDL responds with GENERIC_ERROR, success=false (logs are in attachment)

## Expected result
SDL must respond "UNSUPPORTED_RESOURCE, success=false, info: UI is not supported by system" to mobile app

# Environment
Tag: 4.2.0
commit: a6c32b45832653d6b851be9b19355fa35f377e07
ATF version: 2.2 / 962e1eff8664d97a 

# Attachments
[att_issue#1006.zip](https://github.com/smartdevicelink/sdl_core/files/576176/att_issue.1006.zip)

# Expected delivery   
- [ ] Source code updates 
- [ ] Code comments 
- [ ] UTs add/update 
- [x] ATF tests add/update ***(not required)***  
- [x] Manual tests ***(not required)*** 
- [x] Add/update CI plans/jobs ***(not required)***  
- [x] SDD updates ***(not required)***  
- [x] Guidelines update ([sdl_core_guides](https://github.com/smartdevicelink/sdl_core_guides)) ***(not required)***  
- [x] Guidelines update ([sdl_hmi_integration_guidelines](https://github.com/smartdevicelink/sdl_hmi_integration_guidelines))  ***(not required)***

**Comment by VVeremjova** [comment]
Attached script sends request not only on UI, but on TTS interface too. Attached updated script
[fixed_script.lua.tar.gz](https://github.com/smartdevicelink/sdl_core/files/850040/fixed_script.lua.tar.gz)

**Comment by JackLivio** [comment]
When testing this issue I always receive `UNSUPPORTED_RESOURCE`, but depending on the RPC I either receive differing `SUCCCESS` results.

For example,

Show request response from core
```
{
   "info" : "Destination controller not found!",
   "resultCode" : "UNSUPPORTED_RESOURCE",
   "success" : true
}

```

Alert request response from core
```
{
   "info" : "Destination controller not found!",
   "resultCode" : "UNSUPPORTED_RESOURCE",
   "success" : false
}

```

AddCommand response from core
```
 {
   "info" : "Destination controller not found!",
   "resultCode" : "UNSUPPORTED_RESOURCE",
   "success" : true
}

```

ButtonSubscription Response from core (PRESET_0)
```
{
   "resultCode" : "UNSUPPORTED_RESOURCE",
   "success" : false
}

```

**Comment by LuxoftAKutsan** [comment]
@JackLivio looks like this issue was indirectly fixed with 4028784

**Comment by JackLivio** [comment]
After retesting with isReady:false instead of disabling the ui component on the hmi loading screen, I am able to see that this issue is non reproducible and the correct success cases are sent.
